pybaseball/datahelpers/postprocessing.py

```diff
@@ -18,24 +18,58 @@
     (re.compile(r'^\d{4}-\d{1,2}-\d{1,2}T\d{2}:\d{2}:\d{2}.\d{1,6}Z$'), '%Y-%m-%dT%H:%M:%S.%fZ'),
 ]
 
+
 def try_parse_dataframe(
     data: pd.DataFrame,
     null_replacement: Union[str, int, float, datetime] = np.nan,
     known_percentages: List[str] = []
 ) -> pd.DataFrame:
-    values = [
-        {column: try_parse(data[column][row_i], column) for column in data.columns}
-        for row_i in range(len(data)) 
-    ]
+    data_copy = data.copy()
+    data_copy = coalesce_nulls(data_copy, null_replacement)
+    data_copy = data_copy.apply(pd.to_numeric, errors='ignore', downcast='integer').convert_dtypes(convert_string=False)
+
+    rows = len(data_copy.values)
+
+    for column in data_copy.columns:
+        if str(data_copy[column].dtype) in ['object', 'string']:
```

Comment:
I think you can avoid a level of nesting here if you pre-filter the columns, e.g., 
```
string_columns = [dtype_tuple[0] for dtype_tuple in data_copy.dtypes.items() if str(dtype_tuple[1]) in ["object", "string"]]
for column in string_columns:
  first_value_index = data_copy[column].first_valid_index()
  .
  . etc
```